pkg/vm/stackitem/serialization.go

```diff
@@ -9,6 +9,10 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/io"
 )
 
+// MaxDeserialized is the maximum number one deserialized item can contain
+// (including itself).
+const MaxDeserialized = 2048
```

Comment:
There's one more thing I have doubts about: in C# `MaxItemSize` limit can be set to `data.Length` during `data` stackitem deserialisation (see (these two lines)[https://github.com/neo-project/neo/pull/2531/files#diff-d5debddccfac00a979f666d21d0d83d60de1baae95626d1cf4d6c9e52a3a7ed6L49-R49]), while our implementation always uses `stackitem.MaxSize=1024*1024` constraint which is the default value for `MaxItemSize`.

Although this constraint is used for ByteArray and Buffer deserialisation and I can't imagine the example of a valid ByteArray/Buffer stackitem that has its length in deserialized form exceeding the length in serialized form.

So I think that we're OK with constant `MaxItemSize` constraint.

Reply:
Yes, that's deliberate and it doesn't affect compatibility in any way. Consider some `[BUFFER, 100500, 0]` case with a length of 3 (it's not since 100500 is not 1 byte, but anyway), both implementations will read item type and its length. Then C# implementation will see that `100500 > 3` and error out, our implementation will allocate for new buffer and then try to read 100500 bytes which will fail, so it'll error out too. So the end result is the same and optimizing this error case doesn't worth the trouble. All normal items fit into `MaxItemSize` and thus will be serialized/deserialized without any issues.